DataProcessing/datax-host/src/main/scala/datax/host/BlobBatchingHost.scala

```diff
@@ -124,22 +129,46 @@ object BlobBatchingHost {
 
   // get tracker configs from input arguments and sys env
   // if the trackerFolder is configured, a tracker file should be written in the folder
-  def getTrackerConfigs(config: UnifiedConfig) : (String, Timestamp) = {
+  def getTrackerConfigs(config: UnifiedConfig) : (String, Timestamp, Timestamp) = {
     val inputRoot = config.dict.getOrElse("trackerFolder", "")
     val timeStart = sys.env.getOrElse("process_start_datetime", "")
-    (inputRoot, Timestamp.from(Instant.parse(timeStart)))
+    val timeEnd = sys.env.getOrElse("process_end_datetime", "")
+    (inputRoot, Timestamp.from(Instant.parse(timeStart)), Timestamp.from(Instant.parse(timeEnd)))
   }
 
   // write a tracker file in specific folder with format _SUCCESS_yyyy_MM_dd_HH
-  def writeTracker(trackerFolder: String, dt: Timestamp): Unit = {
+  def writeTracker(trackerFolder: String, dt: Timestamp, dtEnd: Timestamp): Unit = {
     // first create folder if not exists
     HadoopClient.createFolder(trackerFolder)
-    val dateFormat = new SimpleDateFormat("yyyy_MM_dd_HH")
+    var fmt = "yyyy_MM_dd_HH"
+    // if the interval is not 1h, the use the format yyyy_MM_dd_HH_mm
+    if (dtEnd.getTime / 1000 - dt.getTime / 1000 + 1 != 3600) {
```

Comment:
Why have 2 different formats? In case the interval is 1 hour, cant we still have the format as yyyy_MM_dd_HH_mm where mm value will be 00 ?

Reply:
This is to keep the logic on both sides consistent. https://msazure.visualstudio.com/One/_git/DnA-M365A-DataPipeline/pullrequest/6722026?_a=files&path=/src/JS/EdpConfigManager/src/partners/ddvcore/generators/cosmosEgressParamsGenerator.ts